grammar.js

```diff
@@ -775,17 +775,19 @@ module.exports = grammar({
       '}'
     ),
 
-    escape_sequence: $ => token(seq(
+    escape_sequence: $ => prec(1, token(seq(
       '\\',
       choice(
         /u[a-fA-F\d]{4}/,
         /U[a-fA-F\d]{8}/,
         /x[a-fA-F\d]{2}/,
-        /o\d{3}/,
-        /\r\n/,
-        /[^uxo]/
+        /\d{3}/,
+        /\r?\n/,
+        /['"abfrntv\\]/,
       )
-    )),
+    ))),
+
+    _not_escape_sequence: $ => token('\\'),
```

Comment:
I think the `token()` function is not needed here, since '\\' is already a single token; but I could be missing something subtle.

Reply:
you're right